src/views/DashboardView.tsx

```diff
@@ -1,76 +1,148 @@
+import { spacings } from '../components/DesignSystem/DesignSystem';
 import { useAsync } from 'react-use';
 import { get } from '../api/api';
 import useDataGrid from '../hooks/useDatagrid';
-import { rowsWithId } from '../components/DataGrid/utils';
+import Select from '../components/Select/Select';
+import { getLocalStorage, setLocalStorage } from '../utils/localStorage';
+import { uuidExtract } from '../utils/extract';
+
+const columnNames: ColumnTypes[] = ['link', 'title', 'default'];
 
 const DashboardView = () => {
+  const currentCatalogi = getLocalStorage('catalogus')
+    ? getLocalStorage('catalogus')
+    : { value: 'all', label: 'Alle catalogussen' };
+
+  const [catalogus, setCatalogus] = useState<SavedCatalogusT>(currentCatalogi!);
+  const [catalogussen, setCatalogussen] = useState<CatalogusT[]>(undefined!);
   const [query, setQuery] = useState<Query>('');
   const [dataVisualLayout, setDataVisualLayout] = useState<DataVisualLayout>('blocks');
 
   const { loading, value, error } = useAsync(async () => {
-    const results: ZaaktypeT[] = await get(`catalogi/zaaktypen/?status=alles`);
-    return { results };
-  }, []);
+    let endpoint = 'catalogi/zaaktypen/?status=alles';
+    if (catalogus.value !== 'all') endpoint += `&catalogus=${catalogus.value}`;
+    if (!catalogussen) setCatalogussen(await get('catalogi/catalogussen/'));
+    const zaaktypen: ZaaktypeT[] = await get(endpoint);
+
+    return { zaaktypen, catalogussen };
+  }, [catalogus]);
+
+  // Used to show the initial loading skeleton
+  const zaaktypen = value?.zaaktypen
+    ? attributeOnQueryFilter(query, value?.zaaktypen!, 'omschrijving')
+    : new Array(10).fill({});
 
-  const filteredResults = attributeOnQueryFilter(query, value?.results!, 'omschrijving');
-  const zaaktypen: ZaaktypeT[] = !filteredResults ? new Array(10).fill({}) : filteredResults; // Used to show skeleton loading
-  const columnNames: ColumnTypes[] = ['link', 'title', 'default'];
   const initialData = useMemo(
     () => ({
       selection: [],
-      rows: rowsWithId(zaaktypen),
+      rows: zaaktypen?.map((zaaktype, i) => ({
+        ...zaaktype,
+        id: zaaktype.url ? uuidExtract(zaaktype.url) : i,
+      })),
     }),
     [zaaktypen]
   );
+
   const { gridHandlers, gridActionHandlers, ...gridData } = useDataGrid(
     initialData,
     loading,
     columnNames
   );
 
+  const changeCatalogi = (_e: any, child: any) => {
+    const value = {
+      value: child.props.value,
+      label: child.props.children,
+    };
+    setLocalStorage('catalogus', value);
+    setCatalogus(value);
+  };
+
   return (
-    <Stack component={'article'} spacing={spacing} useFlexGap>
-      <Typography variant="h1">{loading ? <Skeleton width={160} /> : 'Dashboard'}</Typography>
+    <Stack component={'article'} spacing={spacings.large} useFlexGap>
+      <Box width={'100%'}>
+        <Typography variant="h1">
+          {loading && !error ? <Skeleton width={160} /> : 'Dashboard'}
```

Comment:
I wonder, is this needed? I find that it is a bit strange that while it's loading, the titles disappear :thinking: (I understand that for the 'Zaaktype' title there is a number next to it, but maybe while it's loading it could display 'Zaaktypen' and once the backend has returned, then the number is added?)

![dashboard-loading](https://github.com/maykinmedia/open-zaaktypebeheer-ui/assets/19154114/20f4e54a-8624-4190-8fcb-6e3b3b2e1dc5)

Reply:
I have done it like this so it would be consistent at the different views.